@archer-65 thanks for taking this on! I gave it a shot and there's a libgccjit issue on my machine (full log below):
```
checking for libgccjit.h... yes
configure: error: The installed libgccjit failed to compile and run a test program using
the libgccjit library; see config.log for the details of the failure.
The test program can be found here:
<https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
You can try compiling it yourself to investigate the issues.
Please report the issue to your distribution if libgccjit was installed
through that.
You can find the instructions on how to compile and install libgccjit from
source on this site:
<https://gcc.gnu.org/wiki/JIT>.
```
(I've not followed those instructions yet.)

My system info:
```
❯ nix --version
nix (Nix) 2.12.0
❯ sw_vers
ProductName:            macOS
ProductVersion:         15.4
BuildVersion:           24E248
```

<details>
<summary>nix log output</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/r4yhgy07cdvsi9n7nn9kc8z9b0kica11-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/wbfdk3yidmz296z9yaxbinl6x4vs6pj4-CVE-2024-53920.patch
patching file lisp/files.el
patching file lisp/progmodes/elisp-mode.el
Hunk #2 succeeded at 473 (offset -28 lines).
applying patch /nix/store/yn26p5niv6yrjhzg4k5rifqlxg9di199-CVE-2025-1244.patch
patching file lisp/man.el
patching file test/lisp/man-tests.el
applying patch /nix/store/llkns5hnmw0aqf7851228gdizlwrdwl7-native-comp-driver-options.patch
patching file lisp/emacs-lisp/comp.el
Hunk #1 succeeded at 186 (offset 8 lines).
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './lisp/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './nt/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './test/infra/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './test/manual/noverlay/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './test/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './msdos/autogen/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './admin/charsets/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './admin/grammars/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './admin/unidata/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './oldXMenu/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './leim/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './mac/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './lib/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './lwlib/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './doc/misc/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './doc/lispref/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './doc/emacs/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './doc/lispintro/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './lib-src/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './nextstep/Makefile.in'
substituteStream() in derivation emacs-mac-macport-29.4: WARNING: pattern /bin/pwd doesn't match anything in file './src/Makefile.in'
Running phase: autoreconfPhase
@nix { "action": "setPhase", "phase": "autoreconfPhase" }
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: /nix/store/n6m6f2nyggkykik90k2kwb7ppm51idjc-autoconf-2.72/bin/autoconf --force
autoreconf: running: /nix/store/n6m6f2nyggkykik90k2kwb7ppm51idjc-autoconf-2.72/bin/autoheader --force
autoreconf: configure.ac: not using Automake
autoreconf: 'build-aux/config.sub' is updated
autoreconf: 'build-aux/config.guess' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/hmffg6n6ylbl4c30pqc9i71mwqzrd0iv-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/34smnas6q9g4i69rqya93znfvcqfgyza-emacs-mac-macport-29.4 --disable-build-details --with-modules --without-gif --without-jpeg --without-png --without-tiff --without-x --without-xpm --enable-mac-app=\$\$out/Applications --with-gnutls --with-mac --with-xml2 --without-ns --with-compress-install --with-toolkit-scroll-bars --with-native-compilation --without-imagemagick --with-mailutils --without-small-ja-dic --with-tree-sitter --without-xinput2 --without-xwidgets --without-dbus --without-selinux
checking for xcrun... xcrun
checking for make... yes
checking for GNU Make... make
checking build system type... aarch64-apple-darwin24.4.0
checking host system type... aarch64-apple-darwin24.4.0
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking whether the compiler is clang... yes
checking for compiler option needed when checking for declarations... -Werror=implicit-function-declaration
checking for ar... ar
checking whether clang and cc understand -c and -o together... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking for linux/fs.h... no
checking for malloc.h... no
checking for sys/systeminfo.h... no
checking for sys/sysinfo.h... no
checking for coff.h... no
checking for pty.h... no
checking for sys/resource.h... yes
checking for sys/utsname.h... yes
checking for pwd.h... yes
checking for utmp.h... yes
checking for util.h... yes
checking for sanitizer/lsan_interface.h... yes
checking for sys/socket.h... yes
checking for sys/param.h... yes
checking for pthread.h... yes
checking for malloc/malloc.h... yes
checking for sys/un.h... yes
checking for vfork.h... no
checking for dirent.h... yes
checking for execinfo.h... yes
checking for stdio_ext.h... no
checking for sys/vfs.h... no
checking for sys/fs_types.h... no
checking for getopt.h... yes
checking for sys/cdefs.h... yes
checking for sys/time.h... yes
checking for ieee754.h... no
checking for limits.h... yes
checking for sys/select.h... yes
checking for stdalign.h... yes
checking for stdbool.h... yes
checking for stdckdint.h... no
checking for sys/random.h... yes
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking how to run the C preprocessor... clang -E
checking for egrep -e... /nix/store/rc5aqp4axf0qdpr45w3m4mmambk230lk-gnugrep-3.11/bin/grep -E
checking for Minix Amsterdam compiler... no
checking for clang option to enable large file support... none needed
checking whether clang accepts -g3 -O2... yes
checking whether the compiler is clang... yes
checking whether C compiler handles -Werror -Wunknown-warning-option... yes
checking whether C compiler handles -Wno-switch... yes
checking whether C compiler handles -Wno-pointer-sign... yes
checking whether C compiler handles -Wno-string-plus-int... yes
checking whether C compiler handles -Wno-unknown-attributes... yes
checking whether C compiler handles -Wno-unknown-pragmas... yes
checking whether C compiler handles -Wno-initializer-overrides... yes
checking whether C compiler handles -Wno-tautological-compare... yes
checking whether C compiler handles -Wno-tautological-constant-out-of-range-compare... yes
checking whether C compiler handles -Wno-deprecated-declarations... yes
checking for a BSD-compatible install... /nix/store/a8lb0q3y83n0bn5s4gayyvfkgjn13ial-coreutils-9.6/bin/install -c
checking command to symlink files in the same directory... ln -s
checking for install-info... /nix/store/arbwd6f5mj6xpjk0hrp8yj9hq1maymxa-texinfo-7.1.1/bin/install-info
checking for gzip... /nix/store/pc04pd0w5a7iidrp5k1q0rfjxa57szqz-gzip-1.13/bin/gzip
checking for 'find' args to delete a file... -delete
checking for brew... no
checking for port... no
checking for -znocombreloc... not needed
checking whether addresses are sanitized... no
checking for math library... none required
checking pkg-config is at least version 0.9.0... yes
checking for machine/soundcard.h... no
checking for sys/soundcard.h... no
checking for soundcard.h... no
checking for mmsystem.h... no
checking for _oss_ioctl in -lossaudio... no
checking for alsa >= 1.0.0... no
checking for ADDR_NO_RANDOMIZE... no
checking for sys/wait.h that is POSIX.1 compatible... yes
checking for net/if.h... yes
checking for ifaddrs.h... yes
checking for net/if_dl.h... yes
checking for struct ifreq.ifr_flags... yes
checking for struct ifreq.ifr_hwaddr... no
checking for struct ifreq.ifr_netmask... no
checking for struct ifreq.ifr_broadaddr... yes
checking for struct ifreq.ifr_addr... yes
checking for struct ifreq.ifr_addr.sa_len... yes
checking whether gcc understands -MMD -MF... yes
checking for X... disabled
checking for Carbon/Carbon.h... yes
checking whether malloc is Doug Lea style... no
checking for sbrk... yes
checking for getpagesize... yes
checking for __lsan_ignore_object... no
checking for fork... yes
checking for vfork... yes
checking for fchmod... yes
checking for canonicalize_file_name... no
checking for faccessat... yes
checking for realpath... yes
checking for lstat... yes
checking for readlinkat... yes
checking for explicit_bzero... no
checking for memset_s... yes
checking for fchmodat... yes
checking for lchmod... yes
checking for fcntl... yes
checking for fdopendir... yes
checking for fstatat... yes
checking for fsync... yes
checking for futimens... yes
checking for getrandom... no
checking for gettimeofday... yes
checking for mkostemp... yes
checking for pipe2... no
checking for pselect... yes
checking for pthread_sigmask... yes
checking for readlink... yes
checking for isblank... yes
checking for iswctype... yes
checking for strtoimax... yes
checking for symlink... yes
checking for localtime_r... yes
checking for timegm... yes
checking for utimensat... yes
checking for getdtablesize... yes
checking for futimes... yes
checking for futimesat... no
checking for lutimes... yes
checking for working mmap... yes
checking for main in -lXbsd... no
checking for pthread library... none needed
checking for thread support... yes
checking for librsvg-2.0 >= 2.14.0... no
checking for libwebpdemux >= 0.6.0... yes
checking for WebPGetInfo... no
checking for libwebpdemux >= 0.6.0 libwebpdecoder >= 0.6.0... yes
checking for sqlite3_open_v2 in -lsqlite3... yes
checking for sqlite3_load_extension in -lsqlite3... yes
checking for getaddrinfo_a in -lanl... no
checking for malloc_trim... no
checking for gnutls >= 2.12.2... yes
checking for libsystemd >= 222... no
checking for jansson >= 2.7... yes
checking for tree-sitter >= 0.20.2... yes
checking for sys/inotify.h... no
checking for libkqueue... no
checking for library containing kqueue... none required
checking for lcms2... no
checking for library containing inflateEnd... -lz
checking for dladdr... yes
checking for dlfunc... no
checking for gcc_jit_context_acquire in -lgccjit... yes
checking for libgccjit.h... yes
configure: error: The installed libgccjit failed to compile and run a test program using
the libgccjit library; see config.log for the details of the failure.
The test program can be found here:
<https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
You can try compiling it yourself to investigate the issues.
Please report the issue to your distribution if libgccjit was installed
through that.
You can find the instructions on how to compile and install libgccjit from
source on this site:
<https://gcc.gnu.org/wiki/JIT>.
```
</details>

--- a/pkgs/applications/editors/emacs/make-emacs.nix
+++ b/pkgs/applications/editors/emacs/make-emacs.nix
@@ -538,10 +538,10 @@ mkDerivation (finalAttrs: {
     };
   };
 
-  meta = meta // {
+  meta = {
     broken = withNativeCompilation && !(stdenv.buildPlatform.canExecute stdenv.hostPlatform);
     knownVulnerabilities = lib.optionals (lib.versionOlder version "30") [
       "CVE-2024-53920 CVE-2025-1244, please use newer versions such as emacs30"
     ];
-  };
+  } // meta;
 })

--- a/pkgs/applications/editors/emacs/sources.nix
+++ b/pkgs/applications/editors/emacs/sources.nix
@@ -152,9 +152,24 @@ in
 
   emacs29-macport = import ./make-emacs.nix (mkArgs {
     pname = "emacs-mac";
-    version = "29.1";
+    version = "29.4";
     variant = "macport";
-    rev = "emacs-29.1-mac-10.0";
-    hash = "sha256-TE829qJdPjeOQ+kD0SfyO8d5YpJjBge/g+nScwj+XVU=";
+    rev = "emacs-29.4-mac-10.1";
+    hash = "sha256-8OQ+fon9tclbh/eUJ09uqKfMaz9M77QnLIp2R8QB6Ic=";
+    patches = fetchpatch: [
+      # CVE-2024-53920
+      (fetchpatch {
+        url = "https://gitweb.gentoo.org/proj/emacs-patches.git/plain/emacs/29.4/07_all_trusted-content.patch?id=f24370de4de0a37304958ec1569d5c50c1745b7f";
+        hash = "sha256-zUWM2HDO5MHEB5fC5TCUxzmSafMvXO5usRzCyp9Q7P4=";
+      })
+
+      # CVE-2025-1244
+      (fetchpatch {
+        url = "https://gitweb.gentoo.org/proj/emacs-patches.git/plain/emacs/29.4/06_all_man.patch?id=f24370de4de0a37304958ec1569d5c50c1745b7f";
+        hash = "sha256-Vdf6GF5YmGoHTkxiD9mdYH0hgvfovZwrqYN1NQ++U1w=";
+      })
+    ];
+
+    meta.knownVulnerabilities = [ ];
   });
 }

--- a/pkgs/applications/editors/emacs/sources.nix
+++ b/pkgs/applications/editors/emacs/sources.nix
@@ -13,6 +13,7 @@ let
       patches ? _: [ ],
       rev,
       hash,
+      meta ? { },
     }:
     {
       inherit
@@ -97,7 +98,7 @@ let
           }
           .${variant};
         mainProgram = "emacs";
-      };
+      } // meta;
     };
 in
 {

